From f97cee993016d7c8ca49fe3548a5d2132e7166ad Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Fri, 12 Jan 2018 22:59:51 +0800
Subject: [PATCH] Strip extra EOL when get log message from file

---
 subversion/svn/util.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/subversion/svn/util.c b/subversion/svn/util.c
index fcd1c2f..0d777b7 100644
--- a/subversion/svn/util.c
+++ b/subversion/svn/util.c
@@ -426,6 +426,21 @@ svn_cl__get_log_message(const char **log_msg,
       truncate_buffer_at_prefix(&(log_msg_str->len), (char *)log_msg_str->data,
                                 EDITOR_EOF_PREFIX);
 
+      /* Strip off extra EOL marker */
+      if (log_msg_str->len >= 1)
+        {
+          if (log_msg_str->len >= 2
+              && log_msg_str->data[log_msg_str->len - 2] == '\r'
+              && log_msg_str->data[log_msg_str->len - 1] == '\n')
+            {
+              log_msg_str->len -= 2;
+              ((char *)log_msg_str->data)[log_msg_str->len] = 0;
+            }
+          else if (log_msg_str->data[log_msg_str->len - 1] == '\r'
+              || log_msg_str->data[log_msg_str->len - 1] == '\n')
+            ((char *)log_msg_str->data)[--log_msg_str->len] = 0;
+        }
+
       *log_msg = log_msg_str->data;
       return SVN_NO_ERROR;
     }
-- 
2.14.1

