From 6d79cf902c7c08ece4abcd26850bb66cfc194304 Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sun, 8 Oct 2017 17:29:29 +0800
Subject: [PATCH 3/3] Add option -p0 to diffstat program

---
 subversion/svn/diff-cmd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/subversion/svn/diff-cmd.c b/subversion/svn/diff-cmd.c
index 3677ad0..be296bf 100644
--- a/subversion/svn/diff-cmd.c
+++ b/subversion/svn/diff-cmd.c
@@ -387,7 +387,7 @@ svn_cl__diff(apr_getopt_t *os,
       apr_procattr_t *attr;
       apr_status_t apr_err;
       const char *progname = "diffstat";
-      const char *args1[] = { progname, "-E", "-C", NULL, };
+      const char *args1[] = { progname, "-E", "-C", "-p0", NULL, };
       const char *args2[] = { progname, NULL, };
 
       apr_err = apr_procattr_create(&attr, pool);
-- 
2.14.1

