From d6ea777825060ed82788bb5a5c755062b31e85be Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sun, 28 Oct 2018 09:35:05 +0800
Subject: [PATCH 4/8] Update README

---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index 6924dff..5323e89 100644
--- a/README
+++ b/README
@@ -1,6 +1,6 @@
 Subversion with my patches, based on 1.9.x
 
-MAIN PATCHES
+Main patches:
       1. Output paging automatically
       2. Ignore files if they match the patterns in .svnignore
       3. Syntax highlight
-- 
1.8.3.1

