From 668722d48041f22c7d929af0127a57b3ec5f480f Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sat, 5 Nov 2016 17:21:02 +0800
Subject: [PATCH 07/95] Ignore nothing in default

---
 subversion/include/svn_config.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/subversion/include/svn_config.h b/subversion/include/svn_config.h
index d194af0..1e060d9 100644
--- a/subversion/include/svn_config.h
+++ b/subversion/include/svn_config.h
@@ -217,9 +217,7 @@ typedef struct svn_config_t svn_config_t;
   "*.rej *~ #*# .#* .*.swp .DS_Store [Tt]humbs.db"
 #endif
 
-#define SVN_CONFIG_DEFAULT_GLOBAL_IGNORES \
-  SVN_CONFIG__DEFAULT_GLOBAL_IGNORES_LINE_1 " " \
-  SVN_CONFIG__DEFAULT_GLOBAL_IGNORES_LINE_2
+#define SVN_CONFIG_DEFAULT_GLOBAL_IGNORES ""
 
 #define SVN_CONFIG_TRUE  "TRUE"
 #define SVN_CONFIG_FALSE "FALSE"
-- 
1.8.3.1

