From 97378b697763498f440986d88e66690c9be1300b Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sat, 5 Nov 2016 20:48:17 +0800
Subject: [PATCH 09/95] Aborting commit automatically if message is empty

---
 subversion/svn/util.c | 35 +++--------------------------------
 1 file changed, 3 insertions(+), 32 deletions(-)

diff --git a/subversion/svn/util.c b/subversion/svn/util.c
index a18e5f3..a27d9dc 100644
--- a/subversion/svn/util.c
+++ b/subversion/svn/util.c
@@ -32,6 +32,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <assert.h>
+#include <stdio.h>
 
 #include <apr_env.h>
 #include <apr_errno.h>
@@ -492,38 +493,8 @@ svn_cl__get_log_message(const char **log_msg,
 
       if (! message)
         {
-          const char *reply;
-          SVN_ERR(svn_cmdline_prompt_user2
-                  (&reply,
-                   _("\nLog message unchanged or not specified\n"
-                     "(a)bort, (c)ontinue, (e)dit:\n"), NULL, pool));
-          if (reply)
-            {
-              int letter = apr_tolower(reply[0]);
-
-              /* If the user chooses to abort, we cleanup the
-                 temporary file and exit the loop with a NULL
-                 message. */
-              if ('a' == letter)
-                {
-                  SVN_ERR(svn_io_remove_file2(lmb->tmpfile_left, FALSE, pool));
-                  *tmp_file = lmb->tmpfile_left = NULL;
-                  break;
-                }
-
-              /* If the user chooses to continue, we make an empty
-                 message, which will cause us to exit the loop.  We
-                 also cleanup the temporary file. */
-              if ('c' == letter)
-                {
-                  SVN_ERR(svn_io_remove_file2(lmb->tmpfile_left, FALSE, pool));
-                  *tmp_file = lmb->tmpfile_left = NULL;
-                  message = svn_stringbuf_create_empty(pool);
-                }
-
-              /* If the user chooses anything else, the loop will
-                 continue on the NULL message. */
-            }
+          fprintf(stdout, "Aborting commit due to empty commit message.\n");
+          break;
         }
     }
 
-- 
1.8.3.1

