From 1900b8793754398bbf29be3fd2230ae347d626b4 Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sun, 2 Apr 2017 13:01:45 +0800
Subject: [PATCH 35/95] Update description of option -v in subcommand commit

---
 subversion/svn/svn.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/subversion/svn/svn.c b/subversion/svn/svn.c
index c4d3df3..c70156e 100644
--- a/subversion/svn/svn.c
+++ b/subversion/svn/svn.c
@@ -677,7 +677,8 @@ const svn_opt_subcommand_desc2_t svn_cl__cmd_table[] =
        "  externals reached by recursion. Do not commit externals with a\n"
        "  fixed revision.\n"),
     {'q', 'N', opt_depth, opt_targets, opt_no_unlock, SVN_CL__LOG_MSG_OPTIONS,
-     opt_changelist, opt_keep_changelists, 'v', opt_include_externals} },
+     opt_changelist, opt_keep_changelists, 'v', opt_include_externals},
+    { 'v', N_("show unified diff that this commit will produce") } },
 
   { "copy", svn_cl__copy, {"cp"}, N_
     ("Copy files and directories in a working copy or repository.\n"
-- 
1.8.3.1

