From a4367874137b87e38f1031b19f8fb94468a274f3 Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sun, 15 Oct 2017 08:54:22 +0800
Subject: [PATCH 58/95] Update README for option --stat

---
 README | 1 +
 1 file changed, 1 insertion(+)

diff --git a/README b/README
index e7d8a4b..e2eab57 100644
--- a/README
+++ b/README
@@ -23,6 +23,7 @@ O.    MAIN PATCHES
       4. New subcommand clean for removing unversioned files
       5. Add option -v for subcommand commit, for print unified diff in log
          message template
+      6. Add option --stat for subcommand diff and log
 
 
 
-- 
1.8.3.1

