From a724e5abd2883b5699614c98089f21c3b4357869 Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Sun, 28 Oct 2018 08:21:48 +0800
Subject: [PATCH 84/95] Set no_peg_revision of do_diff() properly.

---
 subversion/libsvn_client/diff.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/subversion/libsvn_client/diff.c b/subversion/libsvn_client/diff.c
index 374196d..4b8ff56 100644
--- a/subversion/libsvn_client/diff.c
+++ b/subversion/libsvn_client/diff.c
@@ -2774,7 +2774,7 @@ svn_client_diff6_hltb(const apr_array_header_t *options,
   return svn_error_trace(do_diff(NULL, NULL, &dwi.ddi,
                                  path_or_url1, path_or_url2,
                                  revision1, revision2, &peg_revision,
-                                 FALSE, /* no_peg_revision */
+                                 TRUE, /* no_peg_revision */
                                  depth, ignore_ancestry, changelists,
                                  TRUE /* text_deltas */,
                                  diff_processor, ctx, pool, pool));
@@ -3038,7 +3038,7 @@ svn_client_diff_peg6_diffstat(const apr_array_header_t *options,
   SVN_ERR(do_diff(NULL, NULL, &dwi.ddi,
                                  path_or_url, path_or_url,
                                  start_revision, end_revision, peg_revision,
-                                 TRUE, /* no_peg_revision */
+                                 FALSE, /* no_peg_revision */
                                  depth, ignore_ancestry, changelists,
                                  TRUE /* text_deltas */,
                                  diff_processor, ctx, pool, pool));
-- 
1.8.3.1

