diff -uprN subversion-1.9.4/README svn/README
--- subversion-1.9.4/README	2017-10-24 11:43:38.672165302 +0800
+++ svn/README	2017-10-24 07:49:03.000000000 +0800
@@ -23,6 +23,7 @@ O.    MAIN PATCHES
       4. New subcommand clean for removing unversioned files
       5. Add option -v for subcommand commit, for print unified diff in log
          message template
+      6. Add option --stat for subcommand diff and log
 
 
 
diff -uprN subversion-1.9.4/subversion/tests/cmdline/getopt_tests_data/svn_help_log_switch_stdout svn/subversion/tests/cmdline/getopt_tests_data/svn_help_log_switch_stdout
--- subversion-1.9.4/subversion/tests/cmdline/getopt_tests_data/svn_help_log_switch_stdout	2015-05-14 12:00:23.000000000 +0800
+++ svn/subversion/tests/cmdline/getopt_tests_data/svn_help_log_switch_stdout	2017-10-24 07:49:03.000000000 +0800
@@ -101,7 +101,7 @@ Valid options:
   --diff-cmd ARG           : use ARG as diff command
   --internal-diff          : override diff-cmd specified in config file
   -x [--extensions] ARG    : Specify differencing options for external diff or
-                             internal diff or blame. Default: '-u'. Options are
+                             internal diff or blame. Default: '-up'. Options are
                              separated by spaces. Internal diff and blame take:
                                -u, --unified: Show 3 lines of unified context
                                -b, --ignore-space-change: Ignore changes in
@@ -112,6 +112,8 @@ Valid options:
                                -p, --show-c-function: Show C function name
   --search ARG             : use ARG as search pattern (glob syntax)
   --search-and ARG         : combine ARG with the previous search pattern
+  --no-color               : disable syntax highlight
+  --stat                   : display statistics of diff
 
 Global options:
   --username ARG           : specify a username ARG
diff -uprN subversion-1.9.4/subversion/tests/cmdline/getopt_tests_data/svn_help_stdout svn/subversion/tests/cmdline/getopt_tests_data/svn_help_stdout
--- subversion-1.9.4/subversion/tests/cmdline/getopt_tests_data/svn_help_stdout	2017-10-24 11:43:39.804165303 +0800
+++ svn/subversion/tests/cmdline/getopt_tests_data/svn_help_stdout	2017-10-24 07:49:03.000000000 +0800
@@ -15,6 +15,7 @@ Available subcommands:
    cat
    changelist (cl)
    checkout (co)
+   clean
    cleanup
    commit (ci)
    copy (cp)
diff -uprN subversion-1.9.4/subversion/tests/cmdline/getopt_tests_data/svn--help_stdout svn/subversion/tests/cmdline/getopt_tests_data/svn--help_stdout
--- subversion-1.9.4/subversion/tests/cmdline/getopt_tests_data/svn--help_stdout	2017-10-24 11:43:39.803165303 +0800
+++ svn/subversion/tests/cmdline/getopt_tests_data/svn--help_stdout	2017-10-24 07:49:03.000000000 +0800
@@ -15,6 +15,7 @@ Available subcommands:
    cat
    changelist (cl)
    checkout (co)
+   clean
    cleanup
    commit (ci)
    copy (cp)
diff -uprN subversion-1.9.4/subversion/tests/cmdline/getopt_tests.py svn/subversion/tests/cmdline/getopt_tests.py
--- subversion-1.9.4/subversion/tests/cmdline/getopt_tests.py	2015-05-14 12:00:19.000000000 +0800
+++ svn/subversion/tests/cmdline/getopt_tests.py	2017-10-24 07:49:03.000000000 +0800
@@ -83,6 +83,9 @@ del_lines_res = [
                  re.compile(r"\* GPG-Agent"),
                  re.compile(r"\* Mac OS X Keychain"),
                  re.compile(r"\* KWallet \(KDE\)"),
+
+                 # Remove my copyright
+                 re.compile(r".*wuzhouhui.*"),
                 ]
 
 # This is a list of lines to search and replace text on.
diff -uprN subversion-1.9.4/subversion/tests/cmdline/import_tests.py svn/subversion/tests/cmdline/import_tests.py
--- subversion-1.9.4/subversion/tests/cmdline/import_tests.py	2015-02-28 18:37:27.000000000 +0800
+++ svn/subversion/tests/cmdline/import_tests.py	2017-10-24 07:49:03.000000000 +0800
@@ -537,10 +537,10 @@ def import_inherited_ignores(sbox):
      'A    ' + os.path.join(F_path, dir5_path)  + '\n',
      'A    ' + os.path.join(F_path, file5_path) + '\n',
      'A    ' + os.path.join(F_path, dir6_path)  + '\n',
-     'A    ' + os.path.join(F_path, file6_path) + '\n',
      'A    ' + os.path.join(F_path, dir7_path)  + '\n',
      'A    ' + os.path.join(F_path, file7_path) + '\n',
      'A    ' + os.path.join(F_path, dir8_path)  + '\n',
+     'A    ' + os.path.join(F_path, file6_path) + '\n',
      'Updated to revision 5.\n'])
   svntest.actions.run_and_verify_svn(expected_output, [], 'up', wc_dir)
 
